ako-operator/controllers/akoconfig_controller.go

```diff
@@ -0,0 +1,236 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"sync"
+
+	logr "github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	akov1alpha1 "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-operator/api/v1alpha1"
+)
+
+const (
+	CleanupFinalizer = "ako.vmware.com/finalizer"
+)
+
+var (
+	rebootRequired = false
+)
+
+// AKOConfigReconciler reconciles a AKOConfig object
+type AKOConfigReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+var objectList map[types.NamespacedName]runtime.Object
+
+var objListOnce sync.Once
+
+func getObjectList() map[types.NamespacedName]runtime.Object {
+	objListOnce.Do(func() {
+		objectList = make(map[types.NamespacedName]runtime.Object)
+	})
+	return objectList
+}
+
+func finalizerInList(finalizers []string, key string) bool {
+	for _, f := range finalizers {
+		if f == key {
+			return true
+		}
+	}
+	return false
+}
+
+func removeFinalizer(finalizers []string, key string) (result []string) {
+	for _, f := range finalizers {
+		if f == key {
+			continue
+		}
+		result = append(result, f)
+	}
+	return result
+}
+
+// +kubebuilder:rbac:groups=ako.vmware.com,resources=akoconfigs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=ako.vmware.com,resources=akoconfigs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=serviceaccounts,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="apps",resources=statefulsets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="rbac",resources=clusterroles,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="rbac",resources=clusterrolebindings,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="policy",resources=podsecuritypolicies,verbs=get;list;watch;create;update;patch;delete
+
+func (r *AKOConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("ako-operator", req.NamespacedName)
+
+	var ako akov1alpha1.AKOConfig
+	err := r.Client.Get(ctx, req.NamespacedName, &ako)
+	if err != nil && errors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	} else if err != nil {
+		log.Info("unable to fetch AKOConfig object", "err", err)
```

Comment:
The reason I set it to info is because, error prints a traceback. And this condition might happen intermittently, so it leaves unnecessary tracebacks in the logs.